{
  "affected_release": [
    {
      "product_name": "JBoss Core Services",
      "release_date": "2018-07-12T00:00:00Z",
      "advisory": "RHSA-2018:2185",
      "package": "",
      "cpe": "cpe:/a:redhat:jboss_core_services:1::el7"
    },
    {
      "product_name": "JBoss Core Services on RHEL 6",
      "release_date": "2018-07-12T00:00:00Z",
      "advisory": "RHSA-2018:2186",
      "package": "",
      "cpe": "cpe:/a:redhat:jboss_core_services:1::el6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "release_date": "2018-04-10T00:00:00Z",
      "advisory": "RHSA-2018:0998",
      "package": "openssl-1:1.0.2k-12.el7",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat JBoss Core Services 1",
      "release_date": "2018-07-12T00:00:00Z",
      "advisory": "RHSA-2018:2187",
      "package": "",
      "cpe": "cpe:/a:redhat:jboss_core_services:1"
    }
  ],
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 5",
      "fix_state": "Not affected",
      "package_name": "openssl",
      "cpe": "cpe:/o:redhat:enterprise_linux:5"
    },
    {
      "product_name": "Red Hat Enterprise Linux 5",
      "fix_state": "Not affected",
      "package_name": "openssl097a",
      "cpe": "cpe:/o:redhat:enterprise_linux:5"
    },
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Not affected",
      "package_name": "openssl",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Not affected",
      "package_name": "openssl098e",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Not affected",
      "package_name": "openssl098e",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Not affected",
      "package_name": "OVMF",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Not affected",
      "package_name": "openssl",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Virtualization 3",
      "fix_state": "Not affected",
      "package_name": "mingw-virt-viewer",
      "cpe": "cpe:/:redhat:enterprise_linux:::hypervisor"
    },
    {
      "product_name": "Red Hat JBoss Enterprise Application Platform 5",
      "fix_state": "Not affected",
      "package_name": "openssl",
      "cpe": "cpe:/a:redhat:jboss_enterprise_application_platform:5"
    },
    {
      "product_name": "Red Hat JBoss Enterprise Application Platform 6",
      "fix_state": "Will not fix",
      "package_name": "openssl",
      "cpe": "cpe:/a:redhat:jboss_enterprise_application_platform:6"
    },
    {
      "product_name": "Red Hat JBoss Enterprise Web Server 2",
      "fix_state": "Will not fix",
      "package_name": "openssl",
      "cpe": "cpe:/a:redhat:jboss_enterprise_web_server:2"
    },
    {
      "product_name": "Red Hat JBoss Enterprise Web Server 3",
      "fix_state": "Not affected",
      "package_name": "openssl",
      "cpe": "cpe:/a:redhat:jboss_enterprise_web_server:3"
    },
    {
      "product_name": "Red Hat JBoss Web Server 3",
      "fix_state": "Will not fix",
      "package_name": "openssl",
      "cpe": "cpe:/a:redhat:jboss_enterprise_web_server:3"
    }
  ],
  "threat_severity": "Moderate",
  "public_date": "2017-12-07T00:00:00Z",
  "bugzilla": {
    "description": "CVE-2017-3737 openssl: Read/write after SSL object in error state",
    "id": "1523504",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1523504"
  },
  "cvss": {
    "cvss_base_score": "",
    "cvss_scoring_vector": "",
    "status": ""
  },
  "cvss3": {
    "cvss3_base_score": "5.9",
    "cvss3_scoring_vector": "CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:N/A:N",
    "status": "verified"
  },
  "iava": "",
  "cwe": "CWE-391",
  "statement": "",
  "acknowledgement": "",
  "name": "CVE-2017-3737",
  "document_distribution": "",
  "details": [
    "OpenSSL 1.0.2 (starting from version 1.0.2b) introduced an \"error state\" mechanism. The intent was that if a fatal error occurred during a handshake then OpenSSL would move into the error state and would immediately fail if you attempted to continue the handshake. This works as designed for the explicit handshake functions (SSL_do_handshake(), SSL_accept() and SSL_connect()), however due to a bug it does not work correctly if SSL_read() or SSL_write() is called directly. In that scenario, if the handshake fails then a fatal error will be returned in the initial function call. If SSL_read()/SSL_write() is subsequently called by the application for the same SSL object then it will succeed and the data is passed without being decrypted/encrypted directly from the SSL/TLS record layer. In order to exploit this issue an application bug would have to be present that resulted in a call to SSL_read()/SSL_write() being issued after having already received a fatal error. OpenSSL version 1.0.2b-1.0.2m are affected. Fixed in OpenSSL 1.0.2n. OpenSSL 1.1.0 is not affected."
  ],
  "references": [
    "https://www.openssl.org/news/secadv/20171207.txt"
  ]
}